src/components/Item.tsx

```diff
@@ -0,0 +1,143 @@
+import * as React from "react";
+
+// @ts-ignore
+import * as mxGraphJs from "mxgraph-js";
+
+const {
+    mxCell,
+    mxUtils,
+    mxGeometry,
+    mxDragSource,
+    mxGraphHandler,
+  } = mxGraphJs;
+
+import {
+  ImxCell,
+  IMxGraph,
+  IMxGraphContext,
+  MxGraphContext,
+} from "../context/MxGraphContext";
+
+interface IItemProps {
+  text?: string;
+  shape?: string;
+}
+
+interface IItemStatus {
+  container: HTMLDivElement | undefined;
+  text: string;
+  image: string;
+  width: number;
+  height: number;
+  style: string;
```

Comment:
I dont think these values besides "container" should be in state. As they are never dynamically reassigned.